### PR TITLE
feat(reusable-nuget-package-build.yml): adding workflow for building …

### PR DESCRIPTION
**Pull request by SerhiiVybach**
…and pushing NuGet packages

### CODE REVIEW

**Review by David-Pas — APPROVED** [review]

File: .github/workflows/reusable-nuget-package-build.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          ref: ${{ inputs.git-ref }}
+
+      - name: Get commit hash
+        id: get_commit_hash
+        shell: bash
+        run: |
+          echo "commit_hash=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+        
+      - name: NuGet - try to restore from cache
+        uses: actions/cache@v3
```
Comment (povisenko) [inline .github/workflows/reusable-nuget-package-build.yml:36]: if you specifically only want to restore then use `uses: actions/cache/restore@v3`
see https://github.com/CtheSigns/reusable-workflows-service/blob/f6568da09db24e85b58ddadc157df72e5cc598cf/.github/workflows/reusable-ci-gradle.yml#L108

Comment (povisenko) [reply .github/workflows/reusable-nuget-package-build.yml:36]: you could also explicitly look to saving cache  via uses: actions/cache/save@v3

https://github.com/CtheSigns/reusable-workflows-service/blob/f6568da09db24e85b58ddadc157df72e5cc598cf/.github/workflows/reusable-ci-gradle.yml#L126C8-L126C36

File: .github/workflows/reusable-nuget-package-build.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    name: "Build Nuget Package"
+    steps:
+      - name: Clone Repository (Custom Ref)
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.git-ref }}
+
+      - name: Get commit hash
+        id: get_commit_hash
+        shell: bash
```
Comment (povisenko) [inline .github/workflows/reusable-nuget-package-build.yml:31]: I'm sure you could also delete shell: bash and it work

**Review by povisenko — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-nuget-package-build.yml (hunk 1 above)
Comment (SerhiiVybach) [reply .github/workflows/reusable-nuget-package-build.yml:36]: @povisenko, actually `actions/cache@v3` does the same but just saves implicitly as post actions, 
do you think using restore/save will be more convenient?

File: .github/workflows/reusable-nuget-package-build.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        required: true
+      push-package:
+        type: boolean
+        required: false
+        default: false
+        
+jobs:
+  build:
+    runs-on: windows-2019
+    name: "Build Nuget Package"
```
[the remark this answers is not in the document]
Comment (SerhiiVybach) [reply .github/workflows/reusable-nuget-package-build.yml]: thanks, fixed in e681b61ae0591f1a59efb6857b62e6cb00540211

File: .github/workflows/reusable-nuget-package-build.yml (hunk 2 above)
Comment (SerhiiVybach) [reply .github/workflows/reusable-nuget-package-build.yml]: thanks, fixed in e681b61ae0591f1a59efb6857b62e6cb00540211

**Review by povisenko — APPROVED** [review]

File: .github/workflows/reusable-nuget-package-build.yml (hunk 1 above)
Comment (povisenko) [reply .github/workflows/reusable-nuget-package-build.yml:35]: makes sense
approved

**Comment by povisenko** [comment]
@SerhiiVybach please do not forget to squash

**Pull request merged**